src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2109,12 +2109,18 @@ private unsafe ParsingError PrivateParseMinimal()
                             _flags |= Flags.UncPath;
                             idx = i;
                         }
+                        else if (!IsWindowsSystem && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && i - idx == 3 && pUriString[i - 1] == '/')
+                        {
```

Comment:
No localhost was just an example, to be more generic, the URIs of syntax ```file://authority/path?query#fragment``` should not alter backward slashes in path component to forward slashes, irrespective of whether the other components, authority, query, fragment are present or absent.